cola/settings.py

```diff
@@ -60,18 +60,34 @@ def __init__(self, verify=git.is_git_worktree):
 
     def remove_missing(self):
         missing_bookmarks = []
+        update_bookmarks = []
         missing_recent = []
 
         for bookmark in self.bookmarks:
-            if not self.verify(bookmark):
-                missing_bookmarks.append(bookmark)
+            try:
+                if not self.verify(bookmark['path']):
+                    missing_bookmarks.append(bookmark)
+            except:
+                """Old version bookmark format """
+                new_bookmark = { 'path' : bookmark, 'name' : os.path.basename(bookmark) }
+                if not self.verify(bookmark):
+                    missing_bookmarks.append(bookmark)
+                else:
+                    update_bookmarks.append(new_bookmark)
```

Comment:
My comment:

> Two bookmarks version are a list (of different components). self.remove_missing() works well on both settings file versions (on my tests), because it deals with whole unit on list not with content, after this parse, the bookmark will be updated.

was an explanation of behavior of `self.bookmarks.remove(bookmark)` into `def remove_missing(self)`, not for `self.remove_missing()`...